src/App.js

```diff
@@ -2,12 +2,42 @@ import React from 'react';
 
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRed } from './api/goods';
+import { Button } from './Button';
+import { GoodsList } from './GoodsList';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+const buttons = {
+  'Load All goods': getAll,
+  'Load 5 first goods': get5First,
+  'Load red goods': getRed,
+};
+
+class App extends React.Component {
+  state = {
+    goods: [],
+  }
+
+  render() {
+    const { goods } = this.state;
+
+    return (
+      <div className="app">
+        <div className="app__container">
+          {Object.entries(buttons).map(([name, loader]) => (
+            <Button
+              key={name}
+              name={name}
+              handleClick={() => loader().then((goodsFromServer) => {
+                this.setState({ goods: goodsFromServer });
+              })}
```

Comment:
let's create a function which will accept a callback and then set a state with the callback's result
then pass this function and loader to Button and call it from there